src/main/scala/dpla/ingestion3/mappers/providers/MtMapping.scala

```diff
@@ -0,0 +1,169 @@
+package dpla.ingestion3.mappers.providers
+
+import java.net.URI
+
+import dpla.ingestion3.enrichments.normalizations.StringNormalizationUtils._
+import dpla.ingestion3.enrichments.normalizations.filters.{DigitalSurrogateBlockList, ExtentIdentificationList, FormatTypeValuesBlockList}
+import dpla.ingestion3.mappers.utils.{Document, IdMinter, Mapping, XmlExtractor}
+import dpla.ingestion3.messages.{IngestMessage, IngestMessageTemplates, MessageCollector}
+import dpla.ingestion3.model.DplaMapData.{AtLeastOne, ExactlyOne, ZeroToMany, ZeroToOne}
+import dpla.ingestion3.model._
+import dpla.ingestion3.utils.Utils
+import org.json4s.JValue
+import org.json4s.JsonDSL._
+
+import scala.util.{Failure, Success, Try}
+import scala.xml._
+
+
+class MtMapping extends Mapping[NodeSeq] with XmlExtractor with IdMinter[NodeSeq] with IngestMessageTemplates {
+
+  val formatBlockList: Set[String] =
+    DigitalSurrogateBlockList.termList ++
+      ExtentIdentificationList.termList
+
+  // ID minting functions
+  override def useProviderName(): Boolean = true
+
+  override def getProviderName(): String = "mt"
+
+  override def getProviderId(implicit data: Document[NodeSeq]): String =
+    extractString(data \\ "header" \ "identifier").getOrElse(throw new RuntimeException(s"No ID for record $data"))
```

Comment:
Why do you throw a runtime exception here instead of adding an error to the `msgCollector`?

Reply:
The `IdMinter` trait doe not currently extend `IngestMessageTemplates` which it would need to do for the `getProviderId` and other id-related methods in the mappings to log message templates. This gap should probably also be fixed in a different way rather than depending on every mapping to log the message. The better way would probably have the `Mapper` or `Xml/JsonMapper` check for required properties rather than reimplement the same messaging code for every single mapping.

Reply:
:+1: